src/components/Table.tsx

```diff
@@ -77,17 +82,19 @@ export type ColumnDef<TableRowData extends object> = {
   width?: ColumnSize;
 };
 
-export type ElementProps<TableRowData extends object | CustomRowConfig, TableRowKey extends Key> = {
```

Comment:
extra generics was making this type more difficult to parse and we were never injecting unique generics that added any sort of type specificity.

the Key was always being implicitly typed and via the passed in props which would happen regardless if we added a generic for it or not.